surface/src/on_nurbs/fitting_surface_pdm.cpp

```diff
@@ -481,33 +481,19 @@ FittingSurface::initNurbsPCABoundingBox (int order, NurbsDataSurface *m_data, Ei
     Eigen::Vector3d p (eigenvectors_inv * (m_data->interior[i] - mean));
     m_data->interior_param.emplace_back (p (0), p (1));
 
-    if (p (0) > v_max (0))
-      v_max (0) = p (0);
-    if (p (1) > v_max (1))
-      v_max (1) = p (1);
-    if (p (2) > v_max (2))
-      v_max (2) = p (2);
-
-    if (p (0) < v_min (0))
-      v_min (0) = p (0);
-    if (p (1) < v_min (1))
-      v_min (1) = p (1);
-    if (p (2) < v_min (2))
-      v_min (2) = p (2);
+    v_max = v_max.cwiseMax (p);
+    v_min = v_min.cwiseMin (p);
   }
 
-  for (unsigned i = 0; i < s; i++)
+  if (v_max(0) - v_min(0) < std::numeric_limits<double>::epsilon() ||
+      v_max(1) - v_min(1) < std::numeric_limits<double>::epsilon())
```

Comment:
IMO, parentheses would make this clearer, but since subtraction is done before the comparison, the code is still correct.